BenchPress/Generators/AzureDeploymentImporter.cs

```diff
@@ -229,4 +212,132 @@ string outputFolderPath
+    /// <code>
+    /// "format('{0}{1}', "ContosoVar" , "ContosoParam")"
+    /// </code>
+    /// </summary>
+    private static string ResolveParamsAndVariables(
+        string stringToResolve,
+        JsonObject armTemplateObject
+    )
+    {
+        // Find and remove square brackets from the parameter/variable string. Square brackets are specific to
+        // ARM template syntax and are not needed in generated tests.
+        stringToResolve = Regex.Replace(
+            stringToResolve,
+            s_squareBracketPattern,
+            s_squareBracketSubstituition
+        );
+
+        // Find all matches in the parameter/variable string that follows the pattern of "parameters('...')" or
+        // "variables('...')". The regular expression pattern defines two named subexpressions: paramOrVarType, which
+        // represents the type of parameter/variable (e.g., "parameters" or "variables"), and paramOrVarName, which
+        // represents the name of the parameter/variable.
+        var matches = s_parametersOrVariablesRegex.Matches(stringToResolve);
+        foreach (Match match in matches)
+        {
+            var name = match.Groups[s_paramOrVarNameGroupKey].Value;
+            var type = match.Groups[s_paramOrVarTypeGroupKey].Value;
+            var resolvedValue = System.String.Empty;
+
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                // ARM templates will contain a JSON Object representing parameters and variables for the ARM template.
+                // Get the correct JSON Object using the type from the regex match (e.g., "parameters" or "variables").
+                var parametersOrVariablesObj = armTemplateObject[type];
+                if (parametersOrVariablesObj != null)
+                {
+                    // Get the value of the parameter/variable from the JSON Object. If the value is still a JSON
+                    // Object, that means it is a parameter that has a default value, so get the default value (if it
+                    // exists). Otherwise, the value can be converted to a string and assigned to resolveValue.
+                    var resolvedValueNode = parametersOrVariablesObj[name];
+                    if (resolvedValueNode != null)
+                    {
+                        if (resolvedValueNode is JsonObject)
+                        {
+                            resolvedValue = resolvedValueNode[s_defaultValueKey]?.ToString();
+                        }
+                        else
+                        {
+                            resolvedValue = resolvedValueNode.ToString();
+                        }
+                    }
+                }
+            }
+            if (!string.IsNullOrWhiteSpace(resolvedValue))
+            {
+                // Remove any square brackets and replace the match with the resolved value in the original string
+                resolvedValue = Regex.Replace(

```

Comment:
Why would we remove square brackets from the resolved value? If there were square brackets in the resolved value, wouldn't we want to keep them?

Reply:
Similar to what is happening in lines 289-293, we remove the square brackets in the resolved value because square brackets in an ARM template represent an ARM template expression (e.g., `"[concat('storage', uniqueString(resourceGroup().id))]"` or `"[resourceGroup().location]"`). As a result, that syntax in the generated tests would not mean anything to the user which is why they are removed. It's definitely more for "readibility" of the generated tests and to remove as much ARM template specific syntax as possible.

With that said, I could go either way of keeping the square brackets in or not. If we want to keep them in, then I can remove lines 289-293 as well.